Add Options to Postgres Connector?

Hi, this is sorta a question and potentially a connector update request. See below for the answers to the GH issue template, but in short, I'm wondering if it's currently possible to pass postgres connection options?

I'm using the postgres wire protocol, but not an actual postgres server, so it's helpful for configuration to be able to pass options at connection time. E.g. what I do w/ `psycopg2` in python:

```python
conn = psycopg2.connect(
    host=HOST,
    port=5432,
    user=EMAIL,
    password=PASSWORD,
    dbname=DBNAME,
    options="organization=public",
)
```

So basically having the `options` parameter available.

I also quickly tried a URL encoded connection string, but that didn't work. If it's indeed possible, and I'm just dumb, I'll have another look :).

Being able to issue commands at connection-time may also work (e.g. `SET organization ...`).

Thanks!

## What do you use this database for?

Postgres, which is already supported

## Are there any popular and well-maintained clients for this database?

`pg`

## Are you interested in contributing this database connector?

Maybe, if it's easy :)

Thanks -- I tried to test my changes locally as described in [CONTRIBUTING.md](https://github.com/evidence-dev/evidence/blob/next/CONTRIBUTING.md#contribute-code), but am running into:

```
pnpm run dev:example-project

> @ dev:example-project /Users/thauck/personal/code/github.com/tshauck/evidence
> pnpm --filter ./sites/example-project dev


> @evidence-dev/components@3.0.0-usql.27 dev /Users/thauck/personal/code/github.com/tshauck/evidence/sites/example-project
> vite dev --port 3000



  VITE v4.3.9  ready in 878 ms

  ➜  Local:   http://127.0.0.1:3000/
  ➜  Network: use --host to expose
  ➜  press h to show help
/Users/thauck/personal/code/github.com/tshauck/evidence/sites/example-project/src/pages/+layout.js:24
                throw new Error('Unable to load source manifest. Do you need to run build:sources?');
                      ^

Error: Unable to load source manifest. Do you need to run build:sources?
    at Module.loadDB (/Users/thauck/personal/code/github.com/tshauck/evidence/sites/example-project/src/pages/+layout.js:24:9)
/Users/thauck/personal/code/github.com/tshauck/evidence/sites/example-project:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @evidence-dev/components@3.0.0-usql.27 dev: `vite dev --port 3000`
Exit status 1
 ELIFECYCLE  Command failed with exit code 1.
```

Trying what it suggests leads to:

```
$ pnpm run build:sources
 ERR_PNPM_NO_SCRIPT  Missing script: build:sources

Command "build:sources" not found. Did you mean "pnpm run build:docs"?
```

--- a/.changeset/happy-games-wink.md
+++ b/.changeset/happy-games-wink.md
@@ -1,0 +1,5 @@
+---
+'@evidence-dev/postgres': patch
+---
+
+adds options parameter to postgres connection settings

--- a/packages/postgres/index.cjs
+++ b/packages/postgres/index.cjs
@@ -51,7 +51,11 @@ const envMap = {
 		{ key: 'POSTGRES_SCHEMA', deprecated: true },
 		{ key: 'schema', deprecated: true },
 		{ key: 'SCHEMA', deprecated: true }
-	]
+	],
+	options: [
+		{ key: 'EVIDENCE_POSTGRES_OPTIONS', deprecated: false },
+		{ key: 'POSTGRES_OPTIONS', deprecated: false },
+	],
 };
 
 /**
@@ -163,7 +167,8 @@ const runQuery = async (queryString, database, batchSize = 100000) => {
 			password: database ? database.password : getEnv(envMap, 'password'),
 			port: database ? database.port : getEnv(envMap, 'port'),
 			ssl: database ? database.ssl : getEnv(envMap, 'ssl'),
-			connectionString: database ? database.connectionString : getEnv(envMap, 'connString')
+			connectionString: database ? database.connectionString : getEnv(envMap, 'connString'),
+			options: database ? database.options : getEnv(envMap, 'options'),
 		};
 
 		// Override types returned by pg package. The package will return some numbers as strings
@@ -248,6 +253,7 @@ module.exports = runQuery;
  * @property {Object | undefined} ssl
  * @property {string} connectionString
  * @property {string} schema
+ * @property {string} options
  */
 
 /** @type {import('@evidence-dev/db-commons').GetRunner<PostgresOptions>} */
@@ -333,5 +339,11 @@ module.exports.options = {
 		type: 'string',
 		secret: false,
 		description: 'Default schema'
+	},
+	options: {
+		title: 'Options',
+		type: 'string',
+		secret: false,
+		description: 'Connection options'
 	}
 };

--- a/packages/postgres/index.cjs
+++ b/packages/postgres/index.cjs
@@ -54,8 +54,8 @@ const envMap = {
 	],
 	options: [
 		{ key: 'EVIDENCE_POSTGRES_OPTIONS', deprecated: false },
-		{ key: 'POSTGRES_OPTIONS', deprecated: false },
-	],
+		{ key: 'POSTGRES_OPTIONS', deprecated: false }
+	]
 };
 
 /**
@@ -168,7 +168,7 @@ const runQuery = async (queryString, database, batchSize = 100000) => {
 			port: database ? database.port : getEnv(envMap, 'port'),
 			ssl: database ? database.ssl : getEnv(envMap, 'ssl'),
 			connectionString: database ? database.connectionString : getEnv(envMap, 'connString'),
-			options: database ? database.options : getEnv(envMap, 'options'),
+			options: database ? database.options : getEnv(envMap, 'options')
 		};
 
 		// Override types returned by pg package. The package will return some numbers as strings

